### ISSUE TITLE
Reverse sync? From Anki to markdown file?

### ISSUE THREAD

**Issue by vinoff** [lervag/apy#84]
Imagine this scenario:

You like having all your cards on a markdown file, say, anki.md. This is because it is much better to create cards on your computer. It is faster and more comfortable. However, sometimes you are out and about, without access to your computer, and you have an idea for a card, so you add it via ankiDroid. Sadly, this card is now only on AnkiWeb and not on your anki.md file. It would be nice if you also had your card in the anki.md file.

Would it be possible to have a reverse_sync option, in which you merge the anki DB with your anki.md?

What about edits? Imagine you edit a card on AnkiDroid. It would be nice if the edit was propagated to your anki.md file as well.

**Comment by lervag** [comment]
> Imagine this scenario:
> 
> You like having all your cards on a markdown file, say, anki.md. … It would be nice if you also had your card in the anki.md file.

Personally, I think the last sentence is the main "dissonance" here. I don't really see the benefit of having all my cards synchronized to one (or more) markdown files.

I consider my Anki collection to be one part of my "knowledge techique", and my note talking to be another part that is only very loosely coupled. I use Markdown files for my notes (and I have A LOT of them). But they are written in a style that is _not_ suitable for Anki cards. More verbose and with more links - not in a question-and-answer style.

> Would it be possible to have a reverse_sync option, in which you merge the anki DB with your anki.md?
>
> What about edits? Imagine you edit a card on AnkiDroid. It would be nice if the edit was propagated to your anki.md file as well.

I would say almost anything is _possible_, here. But there is a LOT of complicated stuff to make this robust and user friendly. This is something I would never use myself. I also think it is a solution to a problem that you don't really have - IMHO, you should not consider your Anki collection to replace general notes.

As an addendum: I _sometimes_ realize I want to look up something that I only partly remember and that is in my Anki database - and that I don't find in my regular notes. In these cases, I tend to use the `apy list "SEARCH"` and `apy review "SEARCH"` interfaces to find the relevant Anki cards.

**Comment by vinoff** [comment]
I understand.

I just want to clarify that I do not use Anki as a notes alternative. Like you, I also have anki and notes separately. It is just that I like to have my anki cards readily available in a markdown file. This is because that way, I can use NVim and have all my snippets and auto-completion and all that. This makes it easier to add new cards and edit old ones, if I ever find they could be improved. It also serves as a backup of sorts. Redundancy and all that. Having some cards on the markdown and others not feels wrong.

Regardless, thank you for apy again!

**Comment by lervag** [comment]
> I just want to clarify that I do not use Anki as a notes alternative. Like you, I also have anki and notes separately.

Great :)

> It is just that I like to have my anki cards readily available in a markdown file. This is because that way, I can use NVim and have all my snippets and auto-completion and all that. This makes it easier to add new cards and edit old ones, if I ever find they could be improved.

I also use Neovim with snippets and stuff. I use `EDITOR=nvim` and with the `apy review` interface, it is very easy to edit/update/change cards with Neovim. To add cards, I often write things like:

```
topic
Q: question 1
A: answer 2
Q: question 2
A: answer 2
```

Then I use a personal mapping to convert it to MArkdown from Neovim and `apy add-from-file` to add it to Anki.

My relevant Neovim config is here:

* https://github.com/lervag/dotnvim/blob/main/ftplugin/markdown.lua
  * This includes general Markdown stuff, but most of the `<leader>a..` mappings are for apy/Anki.
* https://github.com/lervag/dotnvim/blob/main/UltiSnips/markdown.snippets
* https://github.com/lervag/dotnvim/blob/1c9280ba8d1c8143d488a321c9019ca127ec19a2/autoload/personal/markdown.vim#L131-L308

> It also serves as a backup of sorts. Redundancy and all that. Having some cards on the markdown and others not feels wrong.

Well, I don't think that's so important; or, at least, I've never had the need for this type of backup. When I create new notes, I do as explained above. After `apy add-from-file notes.md`, I do `rm notes.md`. The first times I did it felt wrong, but I realized it is pretty safe as long as I verify that things went well with e.g. `apy review added:1`.
